bridge/src/lib.rs

```diff
@@ -56,13 +56,18 @@ methods!(
     TemporalBridge,
     _rtself, // somehow compiler is sure this is unused and insists on the "_"
 
-    fn create_connection(runtime: AnyObject, host: RString) -> AnyObject {
+    fn create_connection(runtime: AnyObject, host: RString, identity: RString, client_name: RString, client_version: RString) -> AnyObject {
         let host = host.map_err(VM::raise_ex).unwrap().to_string();
         let runtime = runtime.unwrap();
         let runtime = runtime.get_data(&*RUNTIME_WRAPPER);
+        let config = ConnectionConfig {
+            identity: identity.map_err(VM::raise_ex).unwrap().to_string(),
+            client_name: client_name.map_err(VM::raise_ex).unwrap().to_string(),
+            client_version: client_version.map_err(VM::raise_ex).unwrap().to_string(),
+        };
 
         let result = Thread::call_without_gvl(move || {
-            Connection::connect(runtime.tokio_runtime.clone(), host.clone())
+            Connection::connect(runtime.tokio_runtime.clone(), host.clone(), config.clone())
```

Comment:
This is where I would just instantiate the `ClientOptionsBuilder` and pass things through directly